### PR TITLE
Fixing errors found while running O3DE on Android.

### PR DESCRIPTION
**Pull request by moraaar**
Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

File: Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp [hunk 1]
```diff
@@ -25,11 +25,9 @@ namespace AZStd
             // Unimplemented on Android
         }
 
-        void SetThreadPriority(int priority, pthread_attr_t&)
+        void SetThreadPriority([[maybe_unused]] int priority, pthread_attr_t&)
         {
             // (not supported at v r10d)
-            AZ_Warning("System", priority <= 0, "Thread priorities %d not supported on Android!\n", priority);
-            (void)priority;
```
Comment (moraaar) [inline Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp:32]: Removed this spamming message, which doesn't add value at the moment. It'd be better to create a github task instead.

File: Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp [hunk 2]
```diff
@@ -25,11 +25,9 @@ namespace AZStd
             // Unimplemented on Android
         }
 
-        void SetThreadPriority(int priority, pthread_attr_t&)
+        void SetThreadPriority([[maybe_unused]] int priority, pthread_attr_t&)
```
Comment (spham-amzn) [inline Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp:28]: really small nit: (for consistency with the other functions here)
```suggestion
        void SetThreadPriority(int, pthread_attr_t&)
```

**Review by spham-amzn — APPROVED** [review]

File: Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp (hunk 1 above)
Comment (spham-amzn) [reply Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp:32]: This shouldn't have been a warning in the first place. If anything, there should be a PAL 'isThreadPrioritySupported()' function in conjunction with this call instead

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp [hunk 3]
```diff
@@ -56,7 +56,7 @@ namespace AZ
                 AZ::Utils::SetEnv("VK_LAYER_PATH", exeDirectory, 1);
 
                 RawStringList validationLayers = Debug::GetValidationLayers();
-                m_descriptor.m_optionalLayers.insert(m_descriptor.m_requiredLayers.end(), validationLayers.begin(), validationLayers.end());
+                m_descriptor.m_optionalLayers.insert(m_descriptor.m_optionalLayers.end(), validationLayers.begin(), validationLayers.end());
```
Comment (amzn-sj) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp:59]: Nice catch!

**Review by amzn-sj — APPROVED** [review]

**Review by amzn-phist — APPROVED** [review]
Thanks!

**Review by moudgils — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by byrcolin — APPROVED** [review]

File: Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp (hunk 2 above)
Comment (moraaar) [reply Code/Framework/AzCore/Platform/Android/AzCore/std/parallel/internal/thread_Android.cpp:28]: Not worth the cost of another AR :) But if I have to another change in this file I'll do it. Cheers

**Pull request merged**